### PR TITLE
deploy.py: helm3 assertion refactor and bugfix of try/except

### PR DESCRIPTION
**Pull request by consideRatio**
I spotted a non-breaking bug in the try/except logic of the function asserting helm3 is used that I fixed. While doing that I also refactored the helm3 assertion function a bit for readability and added a docstring to the deploy function.

### CODE REVIEW

File: deploy.py [hunk 1]
```diff
@@ -124,33 +124,30 @@ def setup_auth_gcloud(release, cluster=None):
...
+    c = subprocess.run(["helm", "--help"])
+    assert c.returncode == 0, "Helm 3 is required, but helm doesn't seem to be installed!"
+
+    c = subprocess.run(
+        ["helm", "version", "--short"],
+        text=True,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.STDOUT,
     )
+    assert c.returncode == 0 and "v3." in c.stdout, "Helm 3 is required, but another something else seem to be installed!"
```
Comment (sgibson91) [inline deploy.py:138]:
```suggestion
    assert c.returncode == 0 and "v3." in c.stdout, "Helm 3 is required, but a different version seem to be installed!"
```

**Review by sgibson91 — APPROVED** [review]
Thanks @consideRatio! I suggested just a small tweak to an error message.

**Comment by consideRatio** [comment]
@sgibson91 thanks for your review!

It is really nice to address review feedback by agreeing that a commit suggestion make sense and pressing a button :D

**Comment by consideRatio** [comment]
As this PR is approved by @sgibson91 and I have time to verify it doesn't cause issues, I'll go ahead and merge this.

**Pull request merged**